### PR TITLE
Added rudimentary support for literate haskell

### PR DESCRIPTION
**Pull request by dbeecham**
This implements (1) in https://github.com/eagletmt/neco-ghc/issues/47

### CODE REVIEW

File: autoload/necoghc.vim [hunk 1]
```diff
@@ -130,6 +131,10 @@ function! necoghc#get_complete_words(cur_keyword_pos, cur_keyword_str) "{{{
   let l:list = []
   let l:line = getline('.')[: a:cur_keyword_pos]
 
+  if (&filetype == "lhaskell")
```
Comment (Shougo) [inline autoload/necoghc.vim]: Please use `==#` instead of `==`.

File: autoload/necoghc.vim [hunk 2]
```diff
@@ -130,6 +131,10 @@ function! necoghc#get_complete_words(cur_keyword_pos, cur_keyword_str) "{{{
   let l:list = []
   let l:line = getline('.')[: a:cur_keyword_pos]
 
+  if (&filetype == "lhaskell") 
+    let l:line = substitute(l:line, "^>[ \t]*", "", "g")
```
Comment (Shougo) [inline autoload/necoghc.vim]:
```vim
let l:line = substitute(l:line, '^>[ \t]*', '', 'g')
```

is better.

**Comment by Shougo** [comment]
Thanks.  I reviewed it.

**Comment by dbeecham** [comment]
Fixed.

**Comment by Shougo** [comment]
OK.  I merged it.

**Pull request merged**